File: Unit-2/Aditya-Patel-HW2-4.jl
## import packages
using Random
using LinearAlgebra
using DelimitedFiles

## Function to create a square matrix of fractions
function ijmat(n::Integer)
    A = zeros(n,n) 
    for j=1:n 
        for i=1:n 
            A[i,j] = 1/(i+j-1) 
        end 
    end 
    return A
end

## Function to generate a vector of random values
function randVec(n::Integer)
    b = zeros(n,1)
    for i=1:n
        b[i, 1] = randn()
    end
    return b
end

## Write b to a files
writedlm("b-matrix.csv", b, ",")

## solve linear system
A = ijmat(10)
b = randVec(10)

x = A \ b
for val in x
    println(val)
end

File: Unit-2/Aditya-Patel-HW2-4.py
## Import statements
import numpy as np
import scipy as sp

## Function to create the n-by-n A matrix
def ijmat(n:int):
    a = np.zeros((n, n))
    for j in range(0, n):
        for i in range(0, n):
            a[i][j] = 1/(i+j+1) 
    return a

## Read Matrix from files
b = np.loadtxt("b-matrix.csv", delimiter=',')

a = ijmat(10)

## Solve Linear System of Equations
x = np.linalg.solve(a,b)

print("{}x{} Matrix[{}]".format(x.shape[0], 1, type(x[0])))
for val in x:
    print("{}".format(val))